### PR TITLE
sqlcapture: Ignore views and other non-BASE TABLE entities

### PR DESCRIPTION
**Pull request by willdonnelly**
**Description:**

Postgres discovery logic has long ignored views and other entities whose type is not `BASE TABLE`. Unfortunately it looks like MySQL and SQL Server discovery omitted that bit and would happily produce bindings for views, which will never really work properly since they don't have replication events.

This commit changes things so that MySQL and SQL Server discovery should correctly filter these entities, but in such a way that they do not produce an error at capture time for any preexisting captures. Instead views will be silently auto-ignored at capture time.

Hopefully we can unwind some of that logic later on once nobody has any views in their catalogs.

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/connectors/916)

### CODE REVIEW

**Comment by willdonnelly** [comment]
Tests appear to pass on MySQL, but let's wait for the full CI run to complete because this could potentially break Postgres or SQL Server if there's a mistake.

File: source-mysql/discovery.go
Comment (jgraettinger) [inline source-mysql/discovery.go:19]: you intend to eventually get rid of this separate query over tables?

instead of introducing a separate query for tables, could you instead update the columns query to return `table_type` (as you have already done below) and then condition on the column's table type?

File: source-mysql/discovery.go
Comment (jgraettinger) [inline source-mysql/discovery.go:328]: should ColumnInfo below be extended with BaseTable ?

**Review by jgraettinger — APPROVED** [review]
LGTM but wondering if this can be simplified back into a single discovery query

File: source-mysql/discovery.go
Comment (willdonnelly) [reply source-mysql/discovery.go:19]: I started that way initially, but the column-discovery helper function currently returns a list of _column_ information objects and adding an "is this a column of a base table" flag to that felt worse. And since we're doing filtering on both the type _and_ the names of tables now, it felt cleaner to just divide up the responsibilities correctly -- now the column-discovery logic just lists every column of every table, and the table-discovery logic decides what tables we're interested in.

File: source-mysql/discovery.go
Comment (willdonnelly) [reply source-mysql/discovery.go:328]: Ah, I forgot to revert my changes to this query after I decided that just factoring out table-discovery as its own distinct thing was cleaner. I'll revert these edits here.

File: source-mysql/discovery.go
Comment (willdonnelly) [reply source-mysql/discovery.go:19]: To be clear, I think that it makes sense to keep table discovery and column listing distinct inside the database-specific discovery code here going forward, the bits I hope to revert are the existence of the "BaseTable" flag in the resulting discovery info and the conditional logic around auto-ignoring views bound to existing captures.

**Pull request merged**
